ocaml/src/tools/etcd.ml

```diff
@@ -0,0 +1,48 @@
+(*
+Copyright 2015 iNuron NV
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*)
+open Lwt.Infix
+
+let parse_url url =
+  let host,port,path =
+    Scanf.sscanf url "%s@:%i/%s" (fun h i p -> h,i,p)
+  in
+  let peers = [host,port] in
+  peers, path
+
+let retrieve_value peers path =
+  (*
+    curl -X GET "http://127.0.0.1.:5000/v2/keys/path_to/asd/055a61c0_0"
+    or
+    etcdctl --peers=127.0.0.1:5000 get path_to/asd/055a61c0_0
+   *)
+  let peers_s = String.concat "," (List.map (fun (h,p) -> Printf.sprintf "%s:%i" h p) peers) in
+  let cmd = ["etcdctl";
+             "--peers=" ^ peers_s;
+             "get";
+             path
+            ]
+  in
+  let cmd_s = String.concat " " cmd in
+  Lwt_io.eprintlf "cmd_s: %s" cmd_s >>= fun () ->
+  let command = Lwt_process.shell cmd_s in
```

Comment:
fyi `Lwt_process.shell` basically undoes the earlier String.concat.
```
let command = ("", [| "etcdctl"; "--peers="^peers_s; "get"; path |]) in ...
```
should work just as fine